src/jmcomic/jm_downloader.py

```
@@ -50,48 +73,50 @@ class JmDownloader(DownloadCallback):
-        self.execute_by_condition(
+        self.execute_on_condition(
             iter_objs=photo,
-            apply=lambda image: self.download_by_image_detail(image, client),
+            apply=self.download_by_image_detail,
             count_batch=self.option.decide_image_batch_count(photo)
         )
         self.after_photo(photo)
 
-    def download_by_image_detail(self, image: JmImageDetail, client: JmcomicClient):
+    @catch_exception
+    def download_by_image_detail(self, image: JmImageDetail):
```

Ruff (0.8.2)

`JmImageDetail` may be undefined, or defined from star imports

(F405)